### ISSUE TITLE
Redot UI freezes completely: Inconsitencies with the set Rendering Device

### ISSUE THREAD

**Issue by mattepm** [Redot-Engine/redot-engine#758]
### Tested versions

- Reproducible in 4.4 Beta

### System information

Debian 12 - Redot 4.4 beta - Vulkan (Forward+) + CPU Intel i9 - GPU RTX 4060 - NVIDIA-SMI 555.42.06

### Issue description

Redot crashes shortly after creating or opening a project.
Looking at the issue I supposed the issue was connected to Vulkan drivers.

Indeed, playing with the binary args, I found a workaround:

Setting **--rendering-driver opengl3** at startup, Redot executes without crashing. Volumetric fog didn't render properly though, so I proceeded poking a bit more in the settings.
Looking at the driver that was set, it was still Vulkan, so, I proceeded removing the flag "Fallback to Opengl 3", and now everything works fine:
![image](https://github.com/user-attachments/assets/435b84a1-922d-4dba-bd27-3d1311bd0931)

Now the project loads and it works well.
I _guess_ maybe that flag shouldn't be set by default? To me it sounds like a conflict between the Vulkan and the OpenGL drivers, but I am not a Graphics renderer expert.

### Steps to reproduce

1) Open Redot
2) Open a project or create a new one
Redot freezes shortly after, but it freezes faster while navigating over the menus.

Note: the same project works fine in Godot

### Minimal reproduction project (MRP)

[new-game-project.zip](https://github.com/user-attachments/files/17434177/new-game-project.zip)

**Comment by 3ttu** [comment]
This tab is for main renderer (which is vulkan), below is for gl.
It's exactly the same in Godot

**Comment by mattepm** [comment]
> This tab is for main renderer (which is vulkan), below is for gl.
> It's exactly the same in Godot

Yes, the tab is exactly the same. 
What isn't exactly the same is that Redot doesn't open while Godot does.

The only way to make Redot work is to open with the terminal open gl flag, and then to run properly with Vulkan to de-flag the "Fallback to Open GL"

That is the difference, Godot works without all this hassle. Redot doesn't. That is the problem, an average users is left without no clue or way to open Redot without using the terminal.

**Comment by mvaranda** [comment]
## Same to me:

I am running Redot from a fork based on the following commit:
```
commit dda23651e8f4696f7d0d913413149a17f08513d8 (origin/master, origin/HEAD, master)
Merge: 19fb23dbea a03e02442f
Author: RadenTheFolf <57826257+RadenTheFolf@users.noreply.github.com>
Date:   Wed Oct 16 15:32:42 2024 -0400
```

I build the code with the command:
```
scons target=editor platform=linuxbsd  dev_build=yes
```

I am testing my build in 3 machines:
- Machine 1: Mac 27 with native Debian 11, NVIDIA GeForce GTX 775M
- Machine 2: PC with AMD with NVIDIA 1060 with LinuxMint 22
- Machine 3: Macbook Air with native Debian 11 with Intel Graphics

They are all using X11

Machines 1 and 2 have similar random freezing. I do not recollect having problems with machine 3 (Intel). It may suggests correlation with NVIDIA ??? Also, as far as I remember official Godot 4.3 was fine for all 3 machines. Still need more testing to double check the above info.

Following some data collection during freezing in 2 different scenarios for Machine 1:

## Error 1: After Suspend/Wakeup

Attached is the GDB dump of stack traces for all threads after Suspend/Wakeup which also freezes the UI; I need to kill it and while killing it the entire Linux Desktop manager (Cinnamon) freezes for 5~10 seconds until the Redot process closes. Then DM resumes normal.

[freezing-at-wakeup-01.txt](https://github.com/user-attachments/files/17451674/freezing-at-wakeup-01.txt)

GDB code matches with the following source code:
```
URL Fork: https://github.com/mvaranda/redot-engine
commit e054d2012691f09dd43405d113b9321821f0116e (HEAD -> mv, origin/mv)
Date:   Sun Oct 20 14:22:02 2024 -0400

    Instrument OS_Unix::delay_usec to error out and recover in case nanosleep returns wrong remaining value

```

## Error 2: Changing Editor theme
after played TruckTown I open Editor Settings, selected Indigo theme and presses Close. The the UI got frozen. It seems that nanosleep called by OS_Unix::delay_usec returned an odd remaining time very high. I am not sure if the freezing are due to the same problem. Anyways, you will see in my fork I added a protection in OS_Unix::delay_usec. So far I did not see the error message yet. I need to do more testing. Here is the returning with odd value:

```
Thread 23 "redot.linuxbsd." hit Breakpoint 2, OS_Unix::delay_usec (this=0x7ffca13a4510, p_usec=1000) at drivers/unix/os_unix.cpp:291
291		while (nanosleep(&requested, &remaining) == -1 && errno == EINTR) {
(gdb) p p_usec
$1 = 1000
(gdb) p requested
$2 = {tv_sec = 0, tv_nsec = 1000000}
(gdb) p remaining
$3 = {tv_sec = 2, tv_nsec = 140723013437400}
(gdb) 

```

**Comment by mattepm** [comment]
I found that the problem is linked to Xorg... I forced-enabled Wayland on Debian, and now everything works. 
As you know Wayland doesn't support Nvidia GPUs well yet and I don't know how much I am comfortable on using Wayland. But so far it works. Still, the problem is only on Redot. Godot used to work fine, so I guess there's something to tweak in this fork

**Comment by Spartan322** [comment]
Do keep in mind performance can also be dependent on running in a debug mode without optimizations, the 4.4 alpha (it was a "beta" but we demoted to an alpha) was run with this, also @mvaranda you're running even worse then a debug mode because its both debug and dev_build which will cause a massive amount of debug inclusions that absolutely tank performance, please try compiling with `production=yes`

**Comment by mattepm** [comment]
The issue seems solved with the just released 4.3 beta. Thanks to everyone. Closing

**Comment by decryptedchaos** [comment]
> I confirm that the problem is linked to Xorg and Nvidia... I forced-enabled Wayland on Debian, and now everything works. As you know Wayland doesn't support Nvidia GPUs well yet and I don't know how much I am comfortable on using Wayland. But so far it works. Still, the problem is only on Redot. Godot used to work fine, so I guess there's something to tweak in this fork

So,  to  shine some light here, i had a similar issue with 4.4,  but it also happened in Godot 4.4 for me too,  the workaround i used was removing the screen section from the Xorg.conf   your mileage may vary.
